File: src/main/java/assignment/cars/CarDetails.java
package assignment.cars;

import java.util.List;

/**
 * Representation of Car Details.
 *
 * For a summary of car detail {@link Car}
 *
 * @since 1.0
 *
 */
public class CarDetails {
    private Integer id;
    private List<String> bodyTypes;
    private List<String> transmissionTypes;
    private List<String> fuelTypes;

    public CarDetails(Integer id, List<String> bodyTypes, List<String> transmissionTypes, List<String> fuelTypes) {
        this.id = id;
        this.bodyTypes = bodyTypes;
        this.transmissionTypes = transmissionTypes;
        this.fuelTypes = fuelTypes;
    }

    /**
     * Returns the unique car/car detail identifier
     * @return the resource identifier
     */
    public Integer getId() {
        return id;
    }

    public CarDetails setId(Integer id) {
        this.id = id;
        return this;
    }

    /**
     * Returns the list of body types available for the car,
     * example: WAGON, COUPE
     * @return list of car body types
     */
    public List<String> getBodyTypes() {
        return bodyTypes;
    }

    public CarDetails setBodyTypes(List<String> bodyTypes) {
        this.bodyTypes = bodyTypes;
        return this;
    }

    /**
     * Reurns the list of transmission types available for the car,
     * example: AUTO, MANUAL, SPORTS AUTO
     * @return list of car transmission types
     */
    public List<String> getTransmissionTypes() {
        return transmissionTypes;
    }

    public CarDetails setTransmissionTypes(List<String> transmissionTypes) {
        this.transmissionTypes = transmissionTypes;
        return this;
    }

    /**
     * Reurns the list of fuel types available for the car,
     * example: UNLEADED PETROL, DIESEL
     * @return list of car fuel types
     */
    public List<String> getFuelTypes() {
        return fuelTypes;
    }

    public CarDetails setFuelTypes(List<String> fuelTypes) {
        this.fuelTypes = fuelTypes;
        return this;
    }
}

File: client/cars/CarDetails.js
// HINT: Task 2
// We should map response to a class

/*
   A CarDetails class represent an instance of a car's details. This object needs
   to be used in conjunction with a Car object to fully describe a car.
   The following class attributes are mandatory to create an instance of
   CarDetails -
              id : Unique ID identifying the specific car.
              bodyTypes : List of body types of a specific car, e.x Coupe, wagon etc.
              transmissionTypes : List of transmission types, e.x Manual, Auto,
                                Sports Auto etc.
              fuelTypes : List of fuel types the car is designed to work with, e.x
                         Premium Unleaded Petrol, E10, Diesel etc.

    Any changes in the persistent object in the service layer should reflect
    in this class.
*/
//        //Body types: ${this._bodyType}\nTranmissions types: ${this._transmissionType}\nFuel types: ${this._fuelType}`;
'use strict';

export class CarDetails {
    constructor(id, bodyTypes, transmissionTypes, fuelTypes) {
        this._id = id;
        this._bodyTypes = bodyTypes;
        this._transmissionTypes = transmissionTypes;
        this._fuelTypes = fuelTypes;
    }

    get id() {
        return this._id;
    }

    get bodyTypes() {
        return this._bodyTypes;
    }

    get transmissionTypes() {
        return this._transmissionTypes;
    }

    get fuelTypes() {
        return this._fuelTypes;
    }

    getDetails() {
        return `ID: ${this._id} Body types:${this._bodyTypes} Transmission types: ${this._transmissionTypes} Fuel Types: ${this._fuelTypes}`;
    }

}
